app/models/appointment.rb

```diff
@@ -0,0 +1,18 @@
+class Appointment < ApplicationRecord
+  include ActiveModel::Serializers::JSON
+
+  belongs_to :teacher
+  belongs_to :user
+  # validates date to be true and not in the past
+  validates :date, presence: true, date: { after_or_equal_to: Date.today }
+  validates :time, presence: true
+
+  def attributes
```

Comment:
This method doesn't seem to be proper for a model. Models/ActiveRecord is to handle database related logic. This attributes method is better as a helper logic since it's more for presentation than database connection

Reply:
This method has been removed from the model.